tests/test_debug_utils.py

```diff
@@ -0,0 +1,46 @@
+import json
+from typing import TYPE_CHECKING
+
+import pytest
+from algokit_utils._debug_utils import AVMDebuggerSourceMap, PersistSourceMapInput, persist_sourcemaps
+
+from tests.conftest import check_output_stability
+
+if TYPE_CHECKING:
+    from algosdk.v2client.algod import AlgodClient
+
+
+def test_build_teal_sourcemaps(algod_client: "AlgodClient", tmp_path_factory: pytest.TempPathFactory) -> None:
+    cwd = tmp_path_factory.mktemp("cwd")
+
+    approval = """
+#pragma version 6
+int 1
+"""
+
+    clear = """
+#pragma version 6
+int 1
+"""
+    sources = [
+        PersistSourceMapInput(teal=approval, app_name="cool_app", file_name="approval.teal"),
+        PersistSourceMapInput(teal=clear, app_name="cool_app", file_name="clear"),
+    ]
+
+    persist_sourcemaps(sources=sources, project_root=cwd, client=algod_client)
+
+    root_path = cwd / ".algokit" / "sourcemaps"
+    sourcemap_file_path = root_path / "avm.sources"
+    app_output_path = root_path / "cool_app"
+
+    assert (sourcemap_file_path).exists()
+    assert (app_output_path / "approval.teal").exists()
+    assert (app_output_path / "approval.tok.map").exists()
```

Comment:
Do you have any context why the `tok` part is included in the filename? Curious if maybe `.avm.map` or `.teal.map` is better?

JS uses the convention of `code.js` --> `code.js.map`, and uses the same file schema.

Reply:
@neilcampbell thats a convention for file extension for those file when you generate them via goal - but don't think it should be an issue to have a custom extension anyway cause we get it from algod not goal

Reply:
Ah yes. I now remember that you mentioned that the other day :D 
I think that's a decent reason to stay with `tok.map`. Does goal generate as `teal.tok.map` or just `tok.map`?

Reply:
@neilcampbell i believe its {name}.tok.map

Reply:
@aorumbayev I just double checked and it appears goal uses `.teal.tok.map` as the convention, so maybe we standardise on that.

Reply:
Done